packages/angular/src/migrations/update-13-3-0/update-mfe-webpack-config.ts

```diff
@@ -0,0 +1,128 @@
+import { Tree } from '@nrwl/devkit';
+import { logger, formatFiles } from '@nrwl/devkit';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import { forEachExecutorOptions } from '@nrwl/workspace/src/utilities/executor-options-utils';
+import { Identifier, Node, StringLiteral } from 'typescript';
+
+export default async function (tree: Tree) {
+  const NRWL_WEBPACK_BROWSER_BUILDER = '@nrwl/angular:webpack-browser';
+  const CUSTOM_WEBPACK_OPTION = 'customWebpackConfig';
+
+  forEachExecutorOptions(
+    tree,
+    NRWL_WEBPACK_BROWSER_BUILDER,
+    (opts, projectName) => {
+      // Update the webpack config
+      const webpackPath = opts[CUSTOM_WEBPACK_OPTION].path;
+      if (!tree.exists(webpackPath)) {
+        logger.warn(
+          `Webpack config file for project: ${projectName} does not exist. Skipping project.`
+        );
+        return;
+      }
+      const webpackConfig = tree.read(webpackPath, 'utf-8');
+      if (!webpackConfig.includes('ModuleFederationPlugin')) {
+        logger.warn(
+          `Webpack config file for project: ${projectName} is not using Module Federation. Skipping project.`
+        );
+        return;
+      }
+
+      let updatedWebpackFile = addExperimentsObject(webpackConfig);
+      updatedWebpackFile = addLibraryModuleType(updatedWebpackFile);
+      updatedWebpackFile = removeRemoteName(updatedWebpackFile);
+      tree.write(webpackPath, updatedWebpackFile);
+    }
+  );
+
+  await formatFiles(tree);
+}
+
+export function addExperimentsObject(webpackConfig: string) {
+  const WEBPACK_EXPERIMENTS_QUERY =
+    'PropertyAssignment > Identifier[name=experiments] ~ ObjectLiteralExpression';
+  const ast = tsquery.ast(webpackConfig);
+  const webpackExperimentsNode = tsquery(ast, WEBPACK_EXPERIMENTS_QUERY);
+
+  if (webpackExperimentsNode && webpackExperimentsNode.length > 0) {
+    return webpackConfig;
+  }
+
+  const WEBPACK_EXPORT = 'module.exports = {';
+  const openingBracePos =
+    webpackConfig.indexOf(WEBPACK_EXPORT) + WEBPACK_EXPORT.length;
+
+  return `${webpackConfig.slice(0, openingBracePos)}
+  experiments: {
+    outputModule: true  
+  },${webpackConfig.slice(openingBracePos)}`;
+}
+
+export function addLibraryModuleType(webpackConfig: string) {
+  const LIBRARY_NODE_QUERY =
+    'PropertyAssignment > Identifier[name=library] ~ ObjectLiteralExpression';
+  const ast = tsquery.ast(webpackConfig);
+  const libraryModuleNode = tsquery(ast, LIBRARY_NODE_QUERY);
+
+  if (libraryModuleNode && libraryModuleNode.length > 0) {
```

Comment:
I'm not sure if the previous comment also applies here. I don't know if `library` can have other properties inside, I couldn't find any info about it in the Webpack docs. Could you make sure we are not leaving anything out here?

Reply:
Library must be configured with a "type". if it exists, the user has already modified their webpack config. It may be correct for their setup already, I don't want to change that for them if it is.

Reply:
Sure, we don't want to change it if they already changed/added the `type` property. I was just asking you to check if `library` can have other properties inside, because if that's the case, they could have the `library` object with other properties defined inside, but not the `type`. That case is the same as the previous comment.